### PR TITLE
fix(tx-construction): txBuilder now properly redistributes all outputs to first address after de-registration of all stake keys

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

Currently the transaction builder is not redistributing all outputs back to the first address when de-registering all stake keys.

# Proposed Solution

delegatePortfolio should properly move all funds back to the first address when all stake keys are deregistered.

# Important Changes Introduced

delegatePortfolio now uses the greedy input selector if the wallet is currently multi delegating and then decidedes to unregister all stake keys.

### CODE REVIEW

**Review by mkazlauskas — DISMISSED** [review]
LGTM :rocket:

**Review by pczeglik-iohk — APPROVED** [review]
Code looks good. Thanks @AngelCastilloB

File: packages/tx-construction/test/tx-builder/TxBuilderDelegatePortfolio.test.ts [hunk 1]
```diff
@@ -292,9 +294,23 @@ describe('TxBuilder/delegatePortfolio', () => {
       groupedAddresses = txBuilderFactory.groupedAddresses;
       txBuilder = txBuilderFactory.txBuilder;
       txBuilderProviders = txBuilderFactory.txBuilderProviders;
+
+      nonDelegatingWalletTxBuilder = (
+        await createTxBuilder({
+          keyAgent,
+          stakeKeyDelegations: []
```
Comment (mirceahasegan) [inline packages/tx-construction/test/tx-builder/TxBuilderDelegatePortfolio.test.ts:301]: Could you also include 2 unregistered stake keys to the scenario, so we test that it ignores them and uses random improve?

**Review by mirceahasegan — APPROVED** [review]
Great work, @AngelCastilloB ! 🚀

**Review by pczeglik-iohk — APPROVED** [review]

**Pull request merged**